src/linear-client.ts

```diff
@@ -0,0 +1,349 @@
+import { AxiosRequestConfig } from 'axios';
+import { GenericAPIResponse, getBaseRESTInverseUrl, RestClientInverseOptions } from './util/requestUtils';
+import RequestWrapper from './util/requestWrapper';
+import SharedEndpoints from './shared-endpoints';
+
+export class LinearClient extends SharedEndpoints {
+    protected requestWrapper: RequestWrapper;
+
+    /**
+   * @public Creates an instance of the inverse REST API client.
+   *
+   * @param {string} key - your API key
+   * @param {string} secret - your API secret
+   * @param {boolean} [livenet=false]
+   * @param {RestClientInverseOptions} [restInverseOptions={}] options to configure REST API connectivity
+   * @param {AxiosRequestConfig} [requestOptions={}] HTTP networking options for axios
+   */
+
+    constructor(
+        key?: string | undefined,
+        secret?: string | undefined,
+        livenet?: boolean,
+        restInverseOptions:RestClientInverseOptions = {}, // TODO: Rename this type to be more general.
+        requestOptions: AxiosRequestConfig = {}
+      ) {
+        super()
+        this.requestWrapper = new RequestWrapper(
+            key,
+            secret,
+            getBaseRESTInverseUrl(livenet),
+            restInverseOptions,
+            requestOptions
+        );
+        return this;
+      }
+    //------------Market Data Endpoints------------>
```

Comment:
Also a bit of a nitpick but I prefer the previous way of making an obvious category:
```

  /**
   *
   * Market Data Endpoints
   *
   */

```

See how it is currently: https://github.com/tiagosiebler/bybit-api/blob/master/src/rest-client.ts#L36

Reply:
: < no i will assert dominance through my comment styling